ngraph/frontend/paddlepaddle/src/model.cpp

```diff
@@ -228,6 +235,67 @@ namespace ngraph
 #endif
         } // namespace pdpd
 
+        std::vector<std::shared_ptr<OpPlacePDPD>>
+            InputModelPDPD::InputModelPDPDImpl::getOpPlaces() const
+        {
+            if (m_graph_changed)
+            {
+                return determine_cut_nodes();
+            }
+            return m_op_places;
+        }
+
+        std::vector<std::shared_ptr<OpPlacePDPD>>
+            InputModelPDPD::InputModelPDPDImpl::determine_cut_nodes() const
+        {
+            std::queue<OpPlacePDPD*> q;
+            std::unordered_set<OpPlacePDPD*> visited;
+            std::vector<std::shared_ptr<OpPlacePDPD>> new_op_places;
+            new_op_places.reserve(m_op_places.size());
+            // Marking nodes from outputs to inputs/constants
+            for (const auto& output : getOutputs())
+            {
+                if (!output->is_input())
+                {
+                    auto pdpd_output_op =
+                        std::dynamic_pointer_cast<OpPlacePDPD>(output->get_producing_operation());
+                    if (pdpd_output_op && !visited.count(pdpd_output_op.get()))
```

Comment:
do we need this `!visited.count`? It is for case when getOutputs return duplicates?

Reply:
Yes, for the rare or maybe improbable case when several outputs point to same op.

Reply:
Also is check `if (pdpd_output_op)` needed? Should there be an assert instead?